query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/converters/DefaultColumnRequestConverter.java

```diff
@@ -0,0 +1,410 @@
+          builder.append(
+              convertLiteralToString(
+                  convertMapKeyExpressionToLiterals(filter.getRhs()), paramsBuilder, context));
+          builder.append(" IS NULL");
+          break;
+        case CONTAINS_KEYVALUE:
+          List<LiteralConstant> kvp = convertMapKeyValueExpressionToLiterals(filter.getRhs());
+          builder.append(lhs);
+          builder.append("->>");
+          builder.append(convertLiteralToString(kvp.get(MAP_KEY_INDEX), paramsBuilder, context));
+          builder.append(" = ");
+          builder.append(convertLiteralToString(kvp.get(MAP_VALUE_INDEX), paramsBuilder, context));
+          break;
+        case CONTAINS_KEY_LIKE:
+          builder.append(lhs);
+          builder.append("::jsonb::text");
+          builder.append(" ");
+          builder.append(operator);
+          builder.append(" ");
+          builder.append(
+              convertLiteralToString(
+                  convertMapLikeExpressionToLiterals(filter.getRhs()), paramsBuilder, context));
+          break;
+        default:
+          Expression rhs =
+              handleValueConversionForLiteralExpression(filter.getLhs(), filter.getRhs());
+          builder.append(lhs);
+          builder.append(" ");
+          builder.append(operator);
+          builder.append(" ");
+          builder.append(convertExpressionToString(rhs, paramsBuilder, executionContext, context));
+      }
+    }
+    return builder.toString();
+  }
+
+  /**
+   * Handles value conversion of a literal expression based on its associated column.
+   *
+   * @param lhs LHS expression with which literal is associated with
+   * @param rhs RHS expression which needs value conversion if its a literal expression
+   * @return newly created literal {@link Expression} of rhs if converted else the same one.
+   */
+  private Expression handleValueConversionForLiteralExpression(Expression lhs, Expression rhs) {
+    if (!(isSimpleAttributeExpression(lhs) && rhs.getValueCase().equals(LITERAL))) {
+      return rhs;
+    }
+
+    String lhsColumnName = getLogicalColumnName(lhs).orElseThrow(IllegalArgumentException::new);
+    try {
+      Value value =
+          DestinationColumnValueConverter.INSTANCE.convert(
+              rhs.getLiteral().getValue(), tableDefinition.getColumnType(lhsColumnName));
+      return Expression.newBuilder()
+          .setLiteral(LiteralConstant.newBuilder().setValue(value))
+          .build();
+    } catch (Exception e) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Invalid input:{ %s } for bytes column:{ %s }",
+              rhs.getLiteral().getValue(), tableDefinition.getPhysicalColumnName(lhsColumnName)));
+    }
+  }
+
+  private String convertOperatorToString(Operator operator) {
+    switch (operator) {
+      case AND:
+        return "AND";
+      case OR:
+        return "OR";
+      case NOT:
+        return "NOT";
+      case EQ:
+      case CONTAINS_KEY:
```

Comment:
CONTAINS_KEY, CONTAINS_KEYVALUE maps to an empty string, right? As they are not used eventually. It will be easy to read.

Reply:
Done